File: token-validation-spring/src/test/java/no/nav/security/token/support/spring/validation/interceptor/MetaAnnotations.java
package no.nav.security.token.support.spring.validation.interceptor;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;
import java.lang.annotation.ElementType;
import java.lang.annotation.RetentionPolicy;
import no.nav.security.token.support.core.api.Protected;
import no.nav.security.token.support.core.api.ProtectedWithClaims;
import no.nav.security.token.support.core.api.Unprotected;

@Protected
@Target({ ElementType.TYPE, ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)

@interface ProtectedMeta {

}

@ProtectedWithClaims(issuer = "issuer1", claimMap = { "acr=Level4" })
@Target({ ElementType.TYPE, ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)

 @interface ProtectedWithClaimsMeta {

}
@Unprotected
@Target({ ElementType.TYPE, ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)

 @interface UnprotectedMeta {

 }

File: token-validation-spring/src/test/kotlin/no/nav/security/token/support/spring/validation/interceptor/MetaAnnotations.kt
package no.nav.security.token.support.spring.validation.interceptor

import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.ANNOTATION_CLASS
import kotlin.annotation.AnnotationTarget.CLASS
import kotlin.annotation.AnnotationTarget.FUNCTION
import kotlin.annotation.AnnotationTarget.PROPERTY_GETTER
import kotlin.annotation.AnnotationTarget.PROPERTY_SETTER
import no.nav.security.token.support.core.api.Protected
import no.nav.security.token.support.core.api.ProtectedWithClaims
import no.nav.security.token.support.core.api.Unprotected

@Protected
@Target(ANNOTATION_CLASS, CLASS, FUNCTION, PROPERTY_GETTER, PROPERTY_SETTER)
@Retention(RUNTIME)
internal annotation class ProtectedMeta

@ProtectedWithClaims(issuer = "issuer1", claimMap = ["acr=Level4"])
@Target(ANNOTATION_CLASS, CLASS, FUNCTION, PROPERTY_GETTER, PROPERTY_SETTER)
@Retention(RUNTIME)
internal annotation class ProtectedWithClaimsMeta

@Unprotected
@Target(ANNOTATION_CLASS, CLASS, FUNCTION, PROPERTY_GETTER, PROPERTY_SETTER)
@Retention(RUNTIME)
internal annotation class UnprotectedMeta